cpp/tensorrt_llm/batch_manager/kvCacheManager.cpp

```diff
@@ -831,8 +867,20 @@ void WindowBlockManager::allocatePools(bool useUvm)
         {
             poolDtype = nvinfer1::DataType::kINT8;
         }
+        if (pool.containsIndexerKCache)
+        {
+            poolDtype = nvinfer1::DataType::kUINT8;
+        }
 
-        nvinfer1::Dims const cacheShape = ITensor::makeShape({mNumPrimaryBlocks, pool.numLayers, mKVFactor, blockSize});
+        nvinfer1::Dims cacheShape;
+        if (pool.containsIndexerKCache)
+        {
+            cacheShape = ITensor::makeShape({mNumPrimaryBlocks, pool.numLayers, blockSize});
```

Comment:
here, it seems when we slice a layer out of it, the (per-layer) pool would be non-contiguous.